src/lib.rs

```diff
@@ -217,21 +219,29 @@ impl Mnemonic {
 		Mnemonic::from_entropy_in(Language::English, entropy)
 	}
 
-	/// Generate a new [Mnemonic] in the given language.
+	/// Generate a new [Mnemonic] in the given language
+	/// with the given randomness source.
 	/// For the different supported word counts, see documentation on [Mnemonic].
-	#[cfg(feature = "rand")]
-	pub fn generate_in(language: Language, word_count: usize) -> Result<Mnemonic, Error> {
+	pub fn generate_in_with<R>(rng: &mut R, language: Language, word_count: usize) -> Result<Mnemonic, Error>
+		where R: rand_core::RngCore + rand_core::CryptoRng,
+	{
 		if word_count < MIN_NB_WORDS || word_count % 6 != 0 || word_count > MAX_NB_WORDS {
 			return Err(Error::BadWordCount(word_count));
 		}
 
 		let entropy_bytes = (word_count / 3) * 4;
-		let mut rng = rand::thread_rng();
 		let mut entropy = [0u8; (MAX_NB_WORDS / 3) * 4];
-		rand::RngCore::fill_bytes(&mut rng, &mut entropy[0..entropy_bytes]);
+		rand_core::RngCore::fill_bytes(rng, &mut entropy[0..entropy_bytes]);
 		Mnemonic::from_entropy_in(language, &entropy[0..entropy_bytes])
 	}
 
+	/// Generate a new [Mnemonic] in the given language.
+	/// For the different supported word counts, see documentation on [Mnemonic].
+	#[cfg(feature = "rand")]
+	pub fn generate_in(language: Language, word_count: usize) -> Result<Mnemonic, Error> {
```

Comment:
I wonder if this should stay in here for convenience or should be thrown out for increased focus …. If SemVer is a concern: maybe it can be deprecated?

Reply:
eh I think it's ok. I like that you don't have to fiddle with rand to use this.